string source = rogr
string target = battery
time timestamp
string charging_status
int battery_level

